src/python/txtai/pipeline/text/translation.py

```diff
@@ -74,22 +76,28 @@ def __call__(self, texts, target="en", source=None):
         languages = self.detect(values) if not source else [source] * len(values)
         unique = set(languages)
 
-        # Build list of (index, language, text)
-        values = [(x, lang, values[x]) for x, lang in enumerate(languages)]
+        # Build a dict from language to list of (index, text)
+        langdict = defaultdict(list)
+        for x, lang in enumerate(languages):
+            langdict[lang].append((x, values[x]))
 
         results = {}
         for language in unique:
-            # Get all text values for language
-            inputs = [(x, text) for x, lang, text in values if lang == language]
+            # Get all indices and text values for a language
+            inputs = langdict[language]
 
             # Translate text in batches
             outputs = []
             for chunk in self.batch([text for _, text in inputs], self.batchsize):
-                outputs.extend(self.translate(chunk, language, target))
+                outputs.extend(self.translate(chunk, language, target, debug))
 
             # Store output value
             for y, (x, _) in enumerate(inputs):
-                results[x] = outputs[y].strip()
+                if debug:
+                    model, op = outputs[y]
+                    results[x] = (language, model, op.strip())
```

Comment:
I think having translation as the first argument would be most intuitive.

```python
(op.strip(), language, model)
```

Reply:
Done!